frame/contracts/src/weights.rs

```diff
@@ -155,1810 +150,1498 @@ pub struct SubstrateWeight<T>(PhantomData<T>);
-		(74_445_000 as Weight)
-			// Standard Error: 49_000
-			.saturating_add((17_650_000 as Weight).saturating_mul(r as Weight))
+		(57_904_000 as Weight)
+			// Standard Error: 24_000
+			.saturating_add((15_140_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_memory_current(r: u32, ) -> Weight {
-		(54_500_000 as Weight)
-			// Standard Error: 17_000
-			.saturating_add((9_307_000 as Weight).saturating_mul(r as Weight))
+		(52_469_000 as Weight)
+			// Standard Error: 177_000
+			.saturating_add((8_136_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_memory_grow(r: u32, ) -> Weight {
-		(54_382_000 as Weight)
-			// Standard Error: 5_644_000
-			.saturating_add((748_424_000 as Weight).saturating_mul(r as Weight))
+		(53_090_000 as Weight)
+			// Standard Error: 1_323_000
+			.saturating_add((616_067_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i64clz(r: u32, ) -> Weight {
-		(54_133_000 as Weight)
-			// Standard Error: 20_000
-			.saturating_add((15_830_000 as Weight).saturating_mul(r as Weight))
+		(51_250_000 as Weight)
+			// Standard Error: 40_000
+			.saturating_add((2_083_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i64ctz(r: u32, ) -> Weight {
-		(54_129_000 as Weight)
-			// Standard Error: 22_000
-			.saturating_add((15_894_000 as Weight).saturating_mul(r as Weight))
+		(50_908_000 as Weight)
+			// Standard Error: 125_000
+			.saturating_add((2_558_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i64popcnt(r: u32, ) -> Weight {
-		(54_181_000 as Weight)
-			// Standard Error: 22_000
-			.saturating_add((15_847_000 as Weight).saturating_mul(r as Weight))
+		(51_139_000 as Weight)
+			// Standard Error: 51_000
+			.saturating_add((2_102_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i64eqz(r: u32, ) -> Weight {
-		(54_130_000 as Weight)
-			// Standard Error: 17_000
-			.saturating_add((15_825_000 as Weight).saturating_mul(r as Weight))
+		(50_931_000 as Weight)
+			// Standard Error: 34_000
+			.saturating_add((2_372_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i64extendsi32(r: u32, ) -> Weight {
-		(54_122_000 as Weight)
-			// Standard Error: 19_000
-			.saturating_add((15_803_000 as Weight).saturating_mul(r as Weight))
+		(50_615_000 as Weight)
+			// Standard Error: 21_000
+			.saturating_add((2_698_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i64extendui32(r: u32, ) -> Weight {
-		(54_100_000 as Weight)
-			// Standard Error: 28_000
-			.saturating_add((15_822_000 as Weight).saturating_mul(r as Weight))
+		(50_617_000 as Weight)
+			// Standard Error: 169_000
+			.saturating_add((3_449_000 as Weight).saturating_mul(r as Weight))
 	}
 	fn instr_i32wrapi64(r: u32, ) -> Weight {
-		(54_143_000 as Weight)
-			// Standard Error: 19_000
-			.saturating_add((15_868_000 as Weight).saturating_mul(r as Weight))
+		(51_293_000 as Weight)
+			// Standard Error: 38_000
+			.saturating_add((2_013_000 as Weight).saturating_mul(r as Weight))
```

Comment:
The cost of simple instructions was reduced massively. Have you seen such an improvement elsewhere just due to prioritizing the benchbot process @joao-paulo-parity?

Reply:
[The prioritization](https://github.com/paritytech/bench-bot/blob/31ffe6cc6224be11b8a35de9e947d96e1f53b8e4/run#L251) is both I/O scheduling (https://linux.die.net/man/1/ionice) and process scheduling (https://linux.die.net/man/1/nice). I do not understand enough about how the weights are calculated to say if the decreases are justifiable, but more pallets were tested on https://github.com/paritytech/substrate/pull/9675 and you can take a look there.

Reply:
It can't be explained by changes to the bot I think. It does not make sense that the results reported by the `i64eq` benchmark are lower than the ones reported by `i64const` because the `i64eq` benchmark contains more ` i64const` instructions than the `i64const` benchmark itself (in order to load the arguments).

The changes in this PR also shouldn't have changed that.

Reply:
According to [the git history](https://github.com/paritytech/substrate/commits/master/frame/contracts/src/weights.rs), [the last changes to this file](https://github.com/paritytech/substrate/commit/8a88403d3e0791bf2a3a45cfadc202fc85065b7b#diff-ffebabc685eaf81fc15f476b128bc403c9db3d8fd5d36a00de831cfe1c850293) came from https://github.com/paritytech/substrate/pull/9471 so I don't know how they were generated - it could be the case that they were not generated by the bot.

It's also worthwhile to mention the active Rust toolchain: `stable-x86_64-unknown-linux-gnu rustc 1.54.0 (a178d0322 2021-07-26)`. Were the results also using this?

cc @shawntabrizi

Reply:
> So the relative of this two benchmarks against each other shouldn't have been influenced by bot changes.

Sorry, I did not catch that edit before writing the comment above. If it's about the difference between two results reported by the bench-bot as it is now, I do not know why this concern you're pointing out is happening. We could try the following:

1. De-prioritized run for `i64const` in isolation
2. De-prioritized run for `i64eq` in isolation
3. (sanity-check) De-prioritized run for `i64const` and `i64eq` in unison

Then repeat those for prioritized, get all the results and then compare them. Would that be a good approach?

Reply:
I just started a re-run of the bot as a hail marry. Let's see if this result can be reproduced before digging any deeper. If so I don't think it is related to the bot. I would need to check for bugs in my benchmarks.

Reply:
For closure's sake, this was followed-up in https://github.com/paritytech/substrate/pull/9669#discussion_r702983355.